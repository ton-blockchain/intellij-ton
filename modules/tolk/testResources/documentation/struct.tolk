/// Simple struct
struct Point {
    /// X coordinate
    x: int,
    /// Y coordinate with default value
    y: int = 0
}

/// Generic struct
struct Container<T> {
    /// The contained value
    value: T
}

/// Struct with constructor tag
struct(123) TaggedStruct {
    /// Field with complex type
    data: [int, slice]
}

/// Empty struct
struct Empty {
}
