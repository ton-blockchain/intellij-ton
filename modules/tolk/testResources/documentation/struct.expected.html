<!DOCTYPE html>
<html>
<head>
    <meta charset="UTF-8">
    <title>Tolk Documentation Test Results</title>
    <style>
        body { font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif; margin: 20px; background: #fafafa; }
        h1 { color: #2c3e50; border-bottom: 3px solid #3498db; padding-bottom: 10px; }
        .element { margin-bottom: 25px; border: 1px solid #ddd; padding: 20px; background: white; border-radius: 8px; box-shadow: 0 2px 4px rgba(0,0,0,0.1); }
        .element-title { font-weight: bold; color: #2c3e50; margin-bottom: 15px; font-size: 14px; background: #ecf0f1; padding: 8px; border-radius: 4px; }
        .documentation { background: #f8f9fa; padding: 15px; border-radius: 6px; border-left: 4px solid #3498db; font-family: 'JetBrains Mono', 'Fira Code', monospace; line-height: 1.6; }
        
        /* Tolk syntax highlighting */
        .keyword { color: #d73a49; font-weight: bold; }
        .primitive { color: #005cc5; font-weight: 500; }
        .struct { color: #22863a; font-weight: 500; }
        .function { color: #6f42c1; font-weight: 500; }
        .parameter { color: #e36209; }
        .type-alias { color: #d73a49; }
        .type-parameter { color: #6f42c1; font-style: italic; }
        .annotation { color: #6a737d; }
        .string { color: #032f62; }
        .number { color: #005cc5; }
        .comment { color: #6a737d; font-style: italic; }
        .constant { color: #005cc5; font-weight: 500; }
        .global-variable { color: #e36209; font-weight: 500; }
        .identifier { color: #24292e; }
        
        /* Documentation markup */
        .definition-start { border-top: 2px solid #e1e4e8; margin-top: 10px; padding-top: 10px; }
        .definition-end { border-bottom: 1px solid #e1e4e8; margin-bottom: 10px; padding-bottom: 10px; }
        .content-start { margin-top: 15px; }
    </style>
</head>
<body>
    <h1>Tolk Documentation Test Results</h1>
    <div class="element">
        <div class="element-title">Type expression: int</div>
        <div class="documentation">
            <span style="color:#000080;font-weight:bold;">int</span>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Field: x</div>
        <div class="documentation">
            <div class='definition'><pre><span style="color:#000080;font-weight:bold;">struct</span> <span style="color:#a04908;">Point</span>
<span style="color:#660e7a;font-weight:bold;">x</span>: <span style="color:#000080;font-weight:bold;">int</span></pre></div><div class='content'><p>X coordinate</p>
</div>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Type expression: int</div>
        <div class="documentation">
            <span style="color:#000080;font-weight:bold;">int</span>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Field: y</div>
        <div class="documentation">
            <div class='definition'><pre><span style="color:#000080;font-weight:bold;">struct</span> <span style="color:#a04908;">Point</span>
<span style="color:#660e7a;font-weight:bold;">y</span>: <span style="color:#000080;font-weight:bold;">int</span> = <span style="color:#0000ff;">0</span></pre></div><div class='content'><p>Y coordinate with default value</p>
</div>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Struct: Point</div>
        <div class="documentation">
            <div class='definition'><pre>
<span style="color:#000080;font-weight:bold;">struct</span> <span style="color:#a04908;">Point</span><span style=""> {</span>
   <span style="color:#660e7a;font-weight:bold;">x</span>: <span style="color:#000080;font-weight:bold;">int</span>
   <span style="color:#660e7a;font-weight:bold;">y</span>: <span style="color:#000080;font-weight:bold;">int</span> = <span style="color:#0000ff;">0</span>
<span style="">}</span></pre></div><div class='content'><p>Simple struct</p>
</div>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Type param: T</div>
        <div class="documentation">
            <div class='definition'><pre><span style="color:#000080;font-weight:bold;">type parameter</span> <span style="color:#20999d;">T</span></pre></div>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Type expression: T</div>
        <div class="documentation">
            <span style="color:#20999d;">T</span>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Field: value</div>
        <div class="documentation">
            <div class='definition'><pre><span style="color:#000080;font-weight:bold;">struct</span> <span style="color:#a04908;">Container</span>
<span style="color:#660e7a;font-weight:bold;">value</span>: <span style="color:#20999d;">T</span></pre></div><div class='content'><p>The contained value</p>
</div>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Struct: Container</div>
        <div class="documentation">
            <div class='definition'><pre>
<span style="color:#000080;font-weight:bold;">struct</span> <span style="color:#a04908;">Container</span><span style="">&lt;</span><span style="color:#20999d;">T</span><span style="">&gt;</span><span style=""> {</span>
   <span style="color:#660e7a;font-weight:bold;">value</span>: <span style="color:#20999d;">T</span>
<span style="">}</span></pre></div><div class='content'><p>Generic struct</p>
</div>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Type expression: int</div>
        <div class="documentation">
            <span style="color:#000080;font-weight:bold;">int</span>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Type expression: slice</div>
        <div class="documentation">
            <span style="color:#000080;font-weight:bold;">slice</span>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Type expression: [int, slice]</div>
        <div class="documentation">
            <span style="">[</span><span style="color:#000080;font-weight:bold;">int</span><span style="">, </span><span style="color:#000080;font-weight:bold;">slice</span><span style="">]</span>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Field: data</div>
        <div class="documentation">
            <div class='definition'><pre><span style="color:#000080;font-weight:bold;">struct</span> <span style="color:#a04908;">TaggedStruct</span>
<span style="color:#660e7a;font-weight:bold;">data</span>: <span style="">[</span><span style="color:#000080;font-weight:bold;">int</span><span style="">, </span><span style="color:#000080;font-weight:bold;">slice</span><span style="">]</span></pre></div><div class='content'><p>Field with complex type</p>
</div>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Struct: TaggedStruct</div>
        <div class="documentation">
            <div class='definition'><pre>
<span style="color:#000080;font-weight:bold;">struct</span> <span style="">(</span><span style="color:#0000ff;">123</span><span style="">)</span> <span style="color:#a04908;">TaggedStruct</span><span style=""> {</span>
   <span style="color:#660e7a;font-weight:bold;">data</span>: <span style="">[</span><span style="color:#000080;font-weight:bold;">int</span><span style="">, </span><span style="color:#000080;font-weight:bold;">slice</span><span style="">]</span>
<span style="">}</span></pre></div><div class='content'><p>Struct with constructor tag</p>
</div>
        </div>
    </div>
    <div class="element">
        <div class="element-title">Struct: Empty</div>
        <div class="documentation">
            <div class='definition'><pre>
<span style="color:#000080;font-weight:bold;">struct</span> <span style="color:#a04908;">Empty</span><span style=""> {}</span></pre></div><div class='content'><p>Empty struct</p>
</div>
        </div>
    </div>
</body>
</html>
